src/paths/distances.c

```diff
@@ -80,6 +91,19 @@ static int igraph_i_eccentricity(const igraph_t *graph,
             long int dist = igraph_dqueue_long_pop(&q);
             int j, n;
 
+            if (vid_ecc) {
+                /* Return the vertex id of the vertex which has the lowest */
+                /* degree of the vertices most distant from the starting */
+                /* vertex. Assumes there is only 1 vid in vids. Used for */
+                /* pseudo_diameter calculations. */
+                IGRAPH_CHECK(igraph_degree(graph, &degrees, igraph_vss_1(act), mode, IGRAPH_NO_LOOPS));
+                degree = VECTOR(degrees)[0];
```

Comment:
A simple and fast way to query the degree of just one vertex could be through the `adjlist` (the length of adjacency list vectors) instead of through `igraph_degree`

Reply:
I can see now that this is not so easy due to an unfortunate mismatch between the vector type used by `igraph_neighbors` and adjlists: the former is `igraph_real_t` and the latter `igraph_integer_t`.

Reply:
Nevertheless the code does retrieve the degree in both cases, into `n`.